packages/overlay/README.md

```diff
@@ -493,3 +479,119 @@ WebKit clips `position: fixed` elements within containers that have all of:
+            <sp-dialog>
+                <h2 slot="heading" id="outer-dialog-heading">Outer Dialog</h2>
+                <p>This is the outer modal content. Press ESC to close it.</p>
+                <sp-button
+                    id="innerTrigger"
+                    variant="primary"
+                    aria-haspopup="dialog"
+                >
+                    Open Inner Modal
+                </sp-button>
+                <sp-overlay
+                    id="innerOverlay"
+                    type="auto"
+                    trigger="innerTrigger@click"
+                >
+                    <sp-popover>
+                        <sp-dialog>
+                            <h2 slot="heading" id="inner-dialog-heading">
+                                Inner Dialog
+                            </h2>
+                            <p>
+                                This is the inner modal content. Press ESC to
+                                close this first, then the outer modal.
+                            </p>
+                        </sp-dialog>
+                    </sp-popover>
+                </sp-overlay>
+            </sp-dialog>
+        </sp-popover>
+    </sp-overlay>
+</div>
+```
+
+#### Focus management
+
+The overlay manages focus based on its type:
+
+-   For `modal` and `page` types, focus is always trapped within the overlay
+-   For `auto` and `manual` types, focus behavior is controlled by the `receives-focus` attribute
+-   For `hint` type, focus remains on the trigger element
+
+Example of proper focus management:
+
+```html
+<sp-button id="modal-trigger" aria-haspopup="dialog" aria-expanded="false">
+    Open Settings
+</sp-button>
+<sp-overlay trigger="modal-trigger@click" type="modal">
+    <sp-dialog-wrapper
+        headline="Settings"
+        dismissable
+        underlay
+        aria-labelledby="settings-heading"
+    >
+        <h2 id="settings-heading" slot="heading">Settings</h2>
+        <sp-field-label for="setting1">Email Notifications</sp-field-label>
+        <sp-switch id="setting1">Enable notifications</sp-switch>
+
+        <div slot="footer">
+            <sp-button
+                variant="secondary"
+                onclick="this.dispatchEvent(new Event('close', { bubbles: true }))"
+            >
+                Cancel
+            </sp-button>
+            <sp-button
+                variant="accent"
+                onclick="this.dispatchEvent(new Event('close', { bubbles: true }))"
+            >
+                Save
+            </sp-button>
+        </div>
+    </sp-dialog-wrapper>
+</sp-overlay>
+```
+
+#### Keyboard navigation
+
+-   <kbd>ESC</kbd> key closes overlays in reverse order of opening
+-   <kbd>TAB</kbd>/<kbd>Shift+TAB</kbd> navigates through focusable elements within modal/page overlays
+-   <kbd>Arrow keys</kbd> navigate through menu items in menu overlays
+-   <kbd>ENTER</kbd>/<kbd>SPACE</kbd> activates buttons and controls
+
+#### Screen reader considerations
```

Comment:
<3 <3 <3

Reply:
Would love to see this section in all the components incrementally. Great addition. Love how we are making these docs so much self-explanatory.